lib/CPathfinder.h

```diff
@@ -121,4 +176,36 @@ class CPathfinder : private CGameInfoCallback
 	bool addTeleportOneWay(const CGTeleport * obj) const;
 	bool addTeleportOneWayRandom(const CGTeleport * obj) const;
 	bool addTeleportWhirlpool(const CGWhirlpool * obj) const;
+
+	bool canVisitObject() const;
+
+};
+
+struct TurnInfo
+{
+	int turn;
+	int maxMovePointsLand;
+	int maxMovePointsWater;
+	const Bonus * bonusFlying;
+	const Bonus * bonusWaterWalking;
+};
+
+class DLL_LINKAGE CPathfinderHelper
+{
+public:
+	TurnInfo * ti;
```

Comment:
On object created inside getTurnInfo. I'm think you agree that code like "hlp->ti->something" is better than "hlp->ti[turn]->something".

Though I got the idea. Probably I designed it wrong and should store objects themself inside "turnsInfo" instead of storing pointers there.